src/pages/events/home/teams.tsx

```diff
@@ -53,11 +54,28 @@ export const EventTeamsTab: React.FC<EventTagProps> = ({ event }) => {
     return grouped;
   }, [incidents]);
 
+  const filteredTeams = useMemo(() => {
```

Comment:
Do you think it would be helpful to centralize the filtering logic into a single component?